airbyte-integrations/connectors/source-snipeit/source_snipeit/incremental_streams.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC
+from typing import Any, Iterable, MutableMapping, Mapping, Tuple
+from copy import deepcopy
+
+import requests
+import arrow
+from .full_refresh_streams import SnipeitStream
+
+# NOTE: Temporary, dirty hack to account for the import differences
+#       between airbyte_cdk 0.1.53 and 0.1.55
+try:
+    from airbyte_cdk.sources.streams import IncrementalMixin
+except ImportError:
+    from airbyte_cdk.sources.streams.core import IncrementalMixin
+
+class Events(SnipeitStream, IncrementalMixin):
+    primary_key = "id"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._state = {}
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, value):
+        self._state[self.cursor_field] = value
+
+    @property
+    def cursor_field(self):
+        return "updated_at"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # NOTE: This was originally the key function in the max() call below
+        #       I moved it out here to keep mypy happy.
+        def __key_function(item: Tuple) -> Any:
+            return item[1]
+        if current_stream_state == {}:
+            self.state = latest_record[self.cursor_field]
+            return {self.cursor_field: latest_record[self.cursor_field]}
+        else:
+            records = {}
+            records[current_stream_state[self.cursor_field]] = arrow.get(current_stream_state[self.cursor_field])
+            records[latest_record[self.cursor_field]] = arrow.get(latest_record[self.cursor_field])
+            # NOTE: mypy complains about records.items() not having the right type for max() but it works just fine
+            #       in runtime regardless.
+            latest_record = max(records.items(), key=__key_function)[0]   # type: ignore[arg-type]
+            self.state[self.cursor_field] = latest_record
+            return {self.cursor_field: latest_record}
```

Comment:
Please format this function. Also add one unit test for this complex functino.

Reply:
@cornjuliox probably adding a unit test here will help you understand your issue